### ISSUE TITLE
Bug: [prefer-nullish-coalescing]  OR assignment operator(||=) does not report errors or suggest corrections.

### ISSUE THREAD

**Issue by developer-bandi** [typescript-eslint/typescript-eslint#10145]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=CYUwxgNghgTiAEEQBd5QFzwM7JgSwDsBzeAH3gIFcIIBuAKFElgTAHsCd4AjTHfYmQrU69PADN4ACilQypALw8AlMvgBvegF96AdwAWeJNNnyl3VRp1zSinvCA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6MAG1w2HL2jjokNRAC6agL4gjQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare let a: string | null;
declare const b: string | null;
if ((a ||= b)) {
}
while ((a ||= b)) {}
a ||= b
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-nullish-coalescing": [
      "error"
    ]  
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
   "strictNullChecks": true
  }
}
```

### Expected Result

`a ||= b` code has error and convert to `a??=b`

### Actual Result

nothing happens

### Additional Info

I looked at the actual rule code, and I don't see code or test code related to the ||= operator.

I wonder if this requires a rule fix or a document fix.

**Comment by kirkwaiblinger** [comment]
Seems like this was intended to be fixed previously in https://github.com/typescript-eslint/typescript-eslint/issues/5229 and https://github.com/typescript-eslint/typescript-eslint/pull/5234? Wonder if those got lost in a merge or something?

**Comment by developer-bandi** [comment]
I don't know why, but the document in #5234's work is currently merged into the main branch, but the rule code and test code do not seem to be reflected.

When performing this task, I think it would be a good idea to refer to the work in #5234.

**Comment by bradzacher** [comment]
AHHHH I found it.
It looks like this merge commit reverted the changes in the v6 branch for the rule
[`bccf851` (#5886)](https://github.com/typescript-eslint/typescript-eslint/pull/5886/commits/bccf8514621c5d7bc17f1943ed20ec838d32f4dc)

**Comment by JoshuaKGoldberg** [comment]
Nice find! Wild that this lasted so long without discovery. Pity.
